cpp/nacjd.cc

```diff
@@ -0,0 +1,361 @@
+                description += "...";
+            }
+        }
+
+        for (const auto &internal_distribution_node : *distributions_node) {
+            const auto distribution_node(JSON::JSONNode::CastToObjectNodeOrDie("distribution", internal_distribution_node));
+            const auto dates_node(distribution_node->getArrayNode("dates"));
+            const auto licenses_node(distribution_node->getArrayNode("licenses"));
+
+            for (const auto &internal_date_node : *dates_node) {
+                const auto date_node(JSON::JSONNode::CastToObjectNodeOrDie("date", internal_date_node));
+                const auto date_type_node(date_node->getObjectNode("type"));
+                const auto date_date_node(date_node->getStringNode("date"));
+                if (date_type_node->getStringNode("value")->getValue() == "initial release date")
+                    initial_release_date = date_date_node->getValue();
+            }
+
+            for (const auto &internal_license_node : *licenses_node) {
+                const auto license_node(JSON::JSONNode::CastToObjectNodeOrDie("license", internal_license_node));
+                const auto license_name_node(license_node->getStringNode("name"));
+                if (not license_name_node->getValue().empty()) {
+                    license = license_name_node->getValue();
+                    break;
+                }
+            }
+            if (not license.empty() and not initial_release_date.empty())
+                break;
+        }
+
+        if (keywords_node != nullptr) {
+            for (const auto &internal_keyword_node : *keywords_node) {
+                const auto keyword_node(JSON::JSONNode::CastToObjectNodeOrDie("keyword", internal_keyword_node));
+                const auto keyword_value_node(keyword_node->getStringNode("value"));
+                if (not keyword_value_node->getValue().empty())
+                    keywords.emplace(keyword_value_node->getValue());
+            }
+        } else
+            ++no_keywords;
+
+        for (const auto &internal_creator_node : *creators_node) {
+            const auto creator_node(JSON::JSONNode::CastToObjectNodeOrDie("creator", internal_creator_node));
+            const std::string type(creator_node->getStringNode("@type")->getValue());
+            if (type == "Organization") {
+                const std::string name(creator_node->getStringNode("name")->getValue());
+                if (name.empty())
+                    continue;
+                if (ContainsValue(creators, "110"))
+                    creators.emplace(name, "710");
+                else
+                    creators.emplace(name, "110");
+            } else if (type == "Person") {
+                const auto fullName_node(creator_node->getOptionalStringNode("fullName"));
+                if (fullName_node == nullptr)
+                    continue;
+                std::string fullName = fullName_node->getValue();
```

Comment:
Do we also need to normalize author names somehow? Wasn't there something about a function that @ruschein wanted to make re-usable?

Reply:
this ist taken from the json part as (see below).
There is a function called NormalizeAuthorName in convert_json_to_marc to process names with trailing initials, not sure, if you mean this one.
 "creators": [
        {
          "@type": "Person",
          "affiliations": [
            {
              "@type": "Organization",
              "name": "University of Colorado Boulder"
            }
          ],
          "lastName": "Huizinga",
          "firstName": "David",
          **"fullName": "Huizinga, David"**
        }
      ],